ev3dev/core.py

```diff
@@ -605,9 +616,20 @@ def time_sp(self):
     @time_sp.setter
     def time_sp(self, value):
         self.set_attr_int('time_sp', value)
+# ~autogen
 
+    def wait_for_running(self):
+        while True:
+            if 'running' in self.state:
+                return
+            sleep(0.01)
```

Comment:
> I'll throw a counter in so that it raises an Exception if the motor hasn't started after 1s

May be not a counter, but a timeout? And the function could be more generic, something like

```py
def wait_for(condition, timeout=1e5, interval=0.01):
  tic = time.time() + timeout
  done = condition()
  while time.time() < tic and not done:
    sleep(interval)
    done = condition()
  return done
```

Then it could be used as
```py
started = wait_for(lambda: 'running' in motor.state, 1)
stopped = wait_for(lambda: 'running' not in motor.state)
```

Although I am not sure the function belongs to the core ev3dev package. @rhempel, what do you think?